_posts/2024-05-01-develop-guideline-multiple-data-source-in-opensearch-and-plugins.md

```diff
@@ -0,0 +1,341 @@
+
+For additional details on the example plugin code, please refer to this [repo](https://github.com/zhyuanqi/OpenSearch-Dashboards/tree/blog_example/examples/blog_example_plugin). 
+
+## 2 Developing, testing and debugging
+
+Now that we've learned how to integrate multiple data sources with Dashboards plugin, let's move on to the next section and explore some advanced knowledge and skills.
+
+### 2.1 Example Plugins for MDS
+
+To help developers get a quick start with the components, we've included an example plugin for a more intuitive and direct experience. You can run Dashboards with CLI below to enable the example plugins. 
+```
+yarn start --run-examples
+```
+
+Once Dashboards are up and running, navigate to the page and click on the left-side menu. From there, open the Developer examples under OpenSearch Dashboards. Then click on Multiple Data Sources Integration. On the right side of the page, you'll find a comprehensive list of pickers, each accompanied by a brief introduction and a set of properties listed below. Feel free to explore and experiment with the example plugin. 
+
+<img src="/assets/media/blog-images/2024-05-01-develop-guideline-multiple-data-source-in-opensearch-and-plugins/multi_selectable.png">
+
+### 2.2 Data source as a saved object
+
+The data source is persisted as a saved object in the Dashboards system index. At the time of creation, an UUID is assigned as its unique identifier, meta data of the data source is also gathered including remote cluster version, and associated plugins. These autogenerated data as well as user provided information including its title, endpoint, authentication methods, are stored as attributes of the data source object. For current schema of the object, refer to related [code](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data_source/common/data_sources/types.ts#L8). 
+
+## 3 FAQ 
+
+Is any UI component provided by dataSourceManagemet used in core?
+Data source selector is now used in devtools, tutorial, saved object management page. You can see the component after setting `data_source.enabled` to true in `opensearch_dashbards.yml`.
+
+Given that a dashboard is a saved object itself, I assume it will not have a data connection associated with it?
+A data source, represented as a saved object, is identified by its unique saved object ID. For saved objects created from data sources, it has a references section with ID, name and type. This reference helps us to identify the associated data source for the saved object. Here is an example of a index-pattern reference.
+```
+"references": [
+    {
+        "id": "9dc4e190-08a0-11ef-901b-bb0301313422",
+        "type": "data-source",
+        "name": "dataSource"
+    }
+]
+```
+The visualization reference is slightly different as it points to the related index-pattern and use the index-pattern id to find corresponding data source. 
```

Comment:
```suggestion
The visualization reference is slightly different because it points to the related index pattern and uses the index pattern ID to find the corresponding data source: 
```